src/modules/flight_mode_manager/tasks/Auto/FlightTaskAuto.cpp

```diff
@@ -172,25 +177,50 @@ bool FlightTaskAuto::update()
 
 	const bool should_wait_for_yaw_align = _param_mpc_yaw_mode.get() == int32_t(yaw_mode::towards_waypoint_yaw_first)
 					       && !_yaw_sp_aligned;
-	const bool force_zero_velocity_setpoint = should_wait_for_yaw_align || _is_emergency_braking_active;
+
+
 	_updateTrajConstraints();
-	PositionSmoothing::PositionSmoothingSetpoints smoothed_setpoints;
-	_position_smoothing.generateSetpoints(
-		_position,
-		waypoints,
-		_velocity_setpoint,
-		_deltatime,
-		force_zero_velocity_setpoint,
-		smoothed_setpoints
-	);
-
-	_jerk_setpoint = smoothed_setpoints.jerk;
-	_acceleration_setpoint = smoothed_setpoints.acceleration;
-	_velocity_setpoint = smoothed_setpoints.velocity;
-	_position_setpoint = smoothed_setpoints.position;
-
-	_unsmoothed_velocity_setpoint = smoothed_setpoints.unsmoothed_velocity;
-	_want_takeoff = smoothed_setpoints.unsmoothed_velocity(2) < -0.3f;
+
+	if (_stop.isActive()) {
+		_stop.update(_acceleration_setpoint, _velocity_setpoint, _position, _deltatime);
+		_jerk_setpoint = _stop.getJerkSetpoint();
+		_acceleration_setpoint = _stop.getAccelerationSetpoint();
+		_velocity_setpoint = _stop.getVelocitySetpoint();
+		_position_setpoint = _stop.getPositionSetpoint();
+
+
+		_unsmoothed_velocity_setpoint = _stop.getUnsmoothedVelocity();
+		_yaw_setpoint = _stop.getYaw();
+
+	} else if (_stop.wasActive()) {
+		_position_smoothing.reset(_acceleration_setpoint, _velocity, _position);
+
+		// Modifying the triplet outside of the navigatior is not ideal, but this is to catch an edge case.
+		_triplet_prev_wp = _position;
```

Comment:
Important thing to look at during review, Triplets are usually not to be adjusted outside the navigator, but this was the only way I found for the drone to move in a direct line from the end position of the slow-down to the waypoint.